CppCoreGuidelines.md

```diff
@@ -12596,8 +12596,8 @@ Readability. Error prevention. Efficiency.
     for (gsl::index i = 0; i < v.size(); ++i) { // body messes with loop variable: can't be a range-for
         if (i % 2 == 0)
             continue;   // skip even elements
-        else
-            cout << v[i] << '\n';
+
+        cout << v[i] << '\n';
```

Comment:
This feels like going out of your way to use ```continue```. Wouldn't this be more readable as
```
if (i%2 != 0)
   cout << v[i] << "\n";
```

Not sure if my objection is to this code or to F.56.

Doing it the way I suggest doesn't introduce any additional nesting. The maximum depth of the function is still one. The code translates more easily into English ("if i is odd output it to cout with a newline" vs "if i is even then move on. Otherwise output it to cout with a newline"). F.56 says to "Use a guard-clause to take care of exceptional cases and return early." but this hardly seems like an exceptional case (though that is just an example).

However it would appear to violate aspects of F.56. Notably
- "Strive to place the essential code at outermost scope, unless this obscures intent."
- "Flag a functions whose body is simply a conditional statement enclosing a block."

In general I agree with the spirit of F.56 so I'm not sure if this is just taking it to what I'd consider an inappropriate extreme or whether I don't agree with the strictness of the rule in this extreme.

Reply:
I agree with your objection.
I was sort of blinded in my hunt for redundant else.

The code I submitted without really going into depth is an interesting case, but your suggested solution has merits, in that it avoids a `continue`, and still the logic is easy to follow, so I vote for your suggestion.